src/tools/currencyParser/currencyParser.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const DataOrganizer = require('../dataOrganizer/dataOrganizer.js');
+const Table = require('../dataStructures/table.js');
+const fetch = require('../fetch/fetch.js');
+
+class CurrencyParser {
+  constructor(path, updateInterval = false) {
+    this.path = path;
+
+    if (updateInterval) {
+      this.interval = setInterval(() => {
+        this._draw();
+      }, updateInterval);
+    } else {
+      this._draw();
+    }
+  }
```

Comment:
Maybe it would be better if you draw once first, then check for interval.
```suggestion
    this._draw();
    if (updateInterval) {
      this.interval = setInterval(() => {
        this._draw();
      }, updateInterval);
    }
  }
```